Adds lingui translations

p-galligan opened the pull request:
Adds different lingui translations.

p-galligan committed:
add italian

Add italian translation

p-galligan committed:
Add portugese

Adds Brazilian portuguese translation and enhances italian

p-galligan committed:
Adds Spanish

Adds a Spanish translation

p-galligan committed:
Update Spanish

minor changes to Spanish

p-galligan committed:
Add french

Adds French translation

p-galligan committed:
Add German

Adds German translation

p-galligan committed:
Add Japanese

Adds Japanese translation

p-galligan committed:
tweak japanese & german

tweaks translations

p-galligan committed:
Add Korean

Adds Korean translation

p-galligan committed:
Adds Turkey and cleans up

Adds turkey and cleans up other languages.

p-galligan committed:
Add chinese translation and update readme

Updates readme and adds simplified chinese translation

p-galligan committed:
Fix minor translation errors and add message loading

Loads messages for all translations and fixes minor translation errors.

helrond committed:
add locale detection

helrond committed:
implement locale detection

p-galligan committed:
Update translation

Updates chinese translation for one missing string.

p-galligan committed:
Update lingui cli

Updates lingui CLI version

p-galligan committed:
Revert "Update lingui cli"

This reverts commit 9f3180e4e824d7d926ec135885406f384fe2b40d.

p-galligan committed:
update lingui cli

Updates lingui cli version

p-galligan committed:
Merge branch 'development' into translations

p-galligan committed:
Update readme

Updates readme to include changes from dev branch.

p-galligan committed:
Merge branch 'translations' of https://github.com/RockefellerArchiveCenter/dimes into translations

p-galligan commented:
Side note, I’m not sure why the `yarn compile` is failing. Maybe because we changed script names in package.json?

helrond commented on README.md:
```diff
@@ -30,8 +30,37 @@ To add or update reference images, edit the scenarios in `backstop.json` and run
+  * Spanish
+  * Turkish
+
 The repository includes [linguijs](https://lingui.dev/) which is an [Open-source](https://github.com/lingui/js-lingui) Internationalization Framework.  This allows you to do translation work on the UI.
```
Let's move this up as the first paragraph. 

```suggestion
The repository includes [linguijs](https://lingui.dev/) which is an [Open-source](https://github.com/lingui/js-lingui) Internationalization Framework.  This allows you to translate static UI elements into other languages.
```

helrond commented on README.md:
```diff
@@ -30,8 +30,37 @@ To add or update reference images, edit the scenarios in `backstop.json` and run
+  `/src/locales/`. Then add that ISO 639-1 code to the `locales` array in `lingui.config.js`.
+2. Run `$ yarn lingui-extract` to extract translation strings from application code. This should be run every
+  time the HTML structure of the application changes. Failure to run this will result in translated
+  strings being presented as random numbers and letters in the built application. It will automatically generate
```
```suggestion
  strings being presented as random numbers and letters in the built application. This command will automatically generate
```

helrond commented on README.md:
```diff
@@ -30,8 +30,37 @@ To add or update reference images, edit the scenarios in `backstop.json` and run
+
+DIMES uses a [macro implementation](https://lingui.dev/guides/message-extraction#macro-usages) for message extraction. This means that all text with a `<Trans></Trans>` block will be automatically extracted for localization files.
+
+You must update the extractions any time new translation objects or languages are found:
```
```suggestion
You must update files used by lingui any time new translation objects or languages are found:
```

helrond replied:
Wondering if we should break this section up into three different processes:
- Adding a new language
- Updating HTML structure
- Changing translation strings

The commands needed in each case vary, and it might be helpful to have that laid out a bit more clearly. Thoughts?

p-galligan replied:
I think that makes sense! I took a stab at this.

helrond replied:
Looks good! I think we can get rid of that "You must update files used by lingui any time new translation objects or languages are found:" text now because it's covered by the sections below.

p-galligan committed:
Update readme text

Updates text for lingui description

Co-authored-by: Hillel Arnold <helrond@hotmail.com>

p-galligan committed:
Update README.md

Updates build instructions language

Co-authored-by: Hillel Arnold <helrond@hotmail.com>

p-galligan committed:
Update readme

Updates readme to add sections for adding a new language, updating, HTML structure, and changing translation strings.

helrond commented:
> Side note, I’m not sure why the `yarn compile` is failing. Maybe because we changed script names in package.json?

Yup, that's probably why. Looks like the Dockerfile needs to be updated. I'm not sure that compile command should actually be in there?

p-galligan committed:
Update readme

Removes extraneous text.

p-galligan committed:
Update dockerfile and translations

Removes compile from Dockerfile. Adds translation strings.
